java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -257,6 +348,58 @@ public boolean hasNulls() {
   // DATA MOVEMENT
   /////////////////////////////////////////////////////////////////////////////
 
+  private final HostColumnVector.NestedHostColumnVector copyToHostNestedHelper(ColumnViewPointerAccess deviceCvPointer) {
+    // Now some of this we could speed up with caching, but conceptually it should be like the following...
+    // None of these need to be closed because they do not own any data, just views conceptually.
+    if (deviceCvPointer == null) {
+      return null;
+    }
+    HostMemoryBuffer hostOffsets = null;
+    HostMemoryBuffer hostValid = null;
+    HostMemoryBuffer hostData = null;
+    List<HostColumnVector.NestedHostColumnVector> children = new ArrayList<>();
+    MemoryBuffer currData = null;
+    MemoryBuffer currOffsets = null;
+    MemoryBuffer currValidity = null;
+    try {
+
+      long currRows = deviceCvPointer.getNumRows();
+      DType currType = deviceCvPointer.getDataType();
+      currData = deviceCvPointer.getDataBuffer();
+      currOffsets = deviceCvPointer.getOffsetBuffer();
+      currValidity = deviceCvPointer.getValidityBuffer();
+      if (currData != null) {
+        hostData = HostMemoryBuffer.allocate(currData.length);
+        hostData.copyFromDeviceBuffer((DeviceMemoryBufferView)currData);
+      }
+      if (currValidity != null) {
+        hostValid = HostMemoryBuffer.allocate(currValidity.length);
+        hostValid.copyFromDeviceBuffer((DeviceMemoryBufferView)currValidity);
+      }
+      if (currOffsets != null) {
+        hostOffsets = HostMemoryBuffer.allocate(currOffsets.length);
+        hostOffsets.copyFromDeviceBuffer((DeviceMemoryBufferView)currOffsets);
+      }
+      for (int i = 0; i < deviceCvPointer.getNumChildren(); i++) {
+        children.add(copyToHostNestedHelper(deviceCvPointer.getChildColumnView(i)));
+      }
+      // copy offset, valid and data to host as needed
+      // Ownership of children and host buffers is transferred to new HostColumnVectorStandIn
+      return new HostColumnVector.NestedHostColumnVector(currType, currRows, nullCount, hostData,
+          hostValid, hostOffsets, children);
+    } finally {
+      if (currData != null) {
```

Comment:
The host buffers need to be closed on error here too, just like with the TODO down below in the original `copyToHost`

Reply:
done.